packages/flutter_tools/lib/src/build_info.dart

```diff
@@ -247,12 +259,39 @@ class BuildInfo {
     };
   }
 
+
   /// Convert to a structured string encoded structure appropriate for usage as
   /// environment variables or to embed in other scripts.
   ///
   /// Fields that are `null` are excluded from this configuration.
   Map<String, String> toEnvironmentConfig() {
-    return <String, String>{
+    final Map<String, String> result = <String, String>{};
+    if(dartDefineConfigJsonMap != null) {
+      dartDefineConfigJsonMap?.forEach((String key, Object value) {
+        result[key] = '$value';
+      });
+      final List<String> systemVars=<String>[
+        'DART_DEFINES',
+        'DART_OBFUSCATION',
+        'EXTRA_FRONT_END_OPTIONS',
+        'EXTRA_GEN_SNAPSHOT_OPTIONS',
+        'SPLIT_DEBUG_INFO',
+        'TRACK_WIDGET_CREATION',
+        'TREE_SHAKE_ICONS',
+        'PERFORMANCE_MEASUREMENT_FILE',
+        'BUNDLE_SKSL_PATH',
+        'PACKAGE_CONFIG',
+        'CODE_SIZE_DIRECTORY'
+      ];
+      for (final String key in systemVars) {
+        if(result.remove(key) != null){
```

Comment:
I don't think you need this remove, the past implementation you had in which you iterate over result and make an `dartDefineConfigJsonMap.containsKey` is the way to go. I understand that you weren't able to reproduce the dart define but you can mock the `BuildInfo` and send the dart defines there to encode

Reply:
OK,i changed it another way, I don't know if it works